test/specmap/complex.js

```diff
@@ -49,4 +50,17 @@ describe('complex', () => {
       runNextTestCase(0);
     });
   });
+
+  test('should resolve complex specs with allOf and nested references', async () => {
+    // Given
+    const spec = globalThis.loadJsonFile(
+      path.join(__dirname, 'data', 'specs', 'complex-example.json')
+    );
+
+    // When
+    const result = await Swagger.resolve({ spec });
+
+    // Then
+    expect(result).not.toEqual(null);
```

Comment:
Not sure if this is fine or if perhaps we should check some schema here.

Reply:
Matching a snapshot here seems like a bad idea to me as the generated snapshot has a size over a 100 MB.

Reply:
Perhaps we could do a check for `$ref` and `$$ref` to show that the spec is only partially resolved? And the test name should also be changed to `'should partially resolve...'`

```
expect(
      result.spec.components.schemas[
        'com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.StudyTreatments-create'
      ].properties.scenario.allOf[0].$ref
    ).toEqual(
      '#/components/schemas/com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.Scenarios-create'
    );
```

 ```
   expect(
      result.spec.components.schemas[
        'com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.BlindingGroups'
      ].properties.study.properties.scenarios.items.$$ref
    ).toEqual('#/components/schemas/com.sap.ctsm.backend.core.api.study.v1.StudyAPIv1.Scenarios');
```